### PR TITLE
Fix shouldGoBack, gotoPageIndex. Support cmd || ctrl || middle clicks

### PR DESCRIPTION
**Pull request by schwers-zz**

### CODE REVIEW

**Comment by schwers-zz** [comment]
👓 @nramadas

File: lib/modules/shouldGoBack.js [hunk 1]
```diff
@@ -1,11 +1,11 @@
 import { isEqual } from 'lodash/lang';
 
-export default function shouldGoBack(urlHistory, url, queryParams) {
+export default function shouldGoBack(urlHistory, currentIndex, url, queryParams) {
   const existsHistoryAPI = (typeof history !== 'undefined') && history.back && history.state;
-  const existsUrlHistory = urlHistory && urlHistory.length > 1;
+  const existsUrlHistory = urlHistory && urlHistory.length > currentIndex;
```
Comment (nramadas) [inline lib/modules/shouldGoBack.js]: @this works, but doesn't it mean something different?
`const existsUrlHistory = urlHistory && urlHistory.length > 1;` means that the history is atleast 2 in size
`const existsUrlHistory = urlHistory && urlHistory.length > currentIndex;` means that our current index is within the range of history. when would this be false?

File: lib/modules/reducer.js [hunk 2]
```diff
@@ -24,10 +24,11 @@ export default (state=DEFAULT, action={}) => {
     case actions.GOTO_PAGE_INDEX: {
       const { pageIndex } = action.payload;
 
-      return merge(state, {
+      return {
+        ...state,
```
Comment (nramadas) [inline lib/modules/reducer.js:28]: this is gonna rebase conflict. sorry :(

File: lib/modules/components.js [hunk 3]
```diff
@@ -10,6 +10,8 @@ import shouldGoBack from './shouldGoBack';
 
 const T = React.PropTypes;
 
+const isNewTabClick = (e) => e.metaKey || e.ctrlKey || e.button === 1 || e.button === 4;
```
Comment (nramadas) [inline lib/modules/components.js:13]: neat

**Comment by schwers-zz** [comment]
💇

**Comment by nramadas** [comment]
:fish:

**Pull request merged**